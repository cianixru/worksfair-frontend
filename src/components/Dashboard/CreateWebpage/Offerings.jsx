import React from 'react';
import PropTypes from 'prop-types';

import ImageUpload from '../../utils/ImageUpload';
import OfferingsForm from '../../../forms/Webpage/OfferingsForm';
import OfferingItems from './OfferingItem';

const Offerings = ({
  onSubmit,
  validationErrors,
  handleErrorReset,
  handleOfferingImageSelection,
  offerings,
  offeringsFormRef,
  selectedImage,
  handleSaveAndPreview,
  webpage,
  onUpdateOffering,
  handleDelete,
}) => {
  return (
    <div>
      <div className="margin-bottom-25">
        <OfferingItems
          offerings={(webpage && webpage.offerings) || offerings}
          validationErrors={validationErrors}
          handleErrorReset={handleErrorReset}
          handleOfferingImageSelection={handleOfferingImageSelection}
          selectedImage={selectedImage}
          onSubmit={onUpdateOffering}
          colour={webpage && webpage.colour}
          handleDelete={handleDelete}
        />
      </div>
      <div className="columns box margin-bottom-25">
        <div className="column is-one-third">
          <ImageUpload
            handleOfferingImageSelection={handleOfferingImageSelection}
            selectedImage={selectedImage}
          />
        </div>
        <div className="column">
          <OfferingsForm
            onSubmit={onSubmit}
            validationErrors={validationErrors}
            offeringsFormRef={offeringsFormRef}
            handleErrorReset={handleErrorReset}
          />
        </div>
      </div>
      <div className="control">
        <button
          type="submit"
          // disabled={pristine || invalid}
          className="button is-medium is-danger is-rounded"
          data-testid="save-and-preview"
          onClick={handleSaveAndPreview}
        >
          Preview Webpage
        </button>
      </div>
    </div>
  );
};

Offerings.propTypes = {
  onSubmit: PropTypes.func,
  onUpdateOffering: PropTypes.func,
  user: PropTypes.object,
  validationErrors: PropTypes.object,
  handleErrorReset: PropTypes.func,
  handleOfferingImageSelection: PropTypes.func,
  offerings: PropTypes.array,
  offeringsFormRef: PropTypes.object,
  selectedImage: PropTypes.string,
  handleSaveAndPreview: PropTypes.func,
  handleDelete: PropTypes.func,
  webpage: PropTypes.object,
};

export default Offerings;
