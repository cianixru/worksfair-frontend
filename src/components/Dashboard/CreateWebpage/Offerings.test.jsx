import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup, fireEvent } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import Offerings from './Offerings';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { newWebpage: {} }
});

describe('Offerings.jsx', () => {
  const onSubmit = jest.fn();
  const user = {};
  const validationErrors = {};
  const handleErrorReset = jest.fn();
  const handleOfferingImageSelection = jest.fn();
  const offerings = [];
  const selectedImage = 'dahehjsdsa.ocm';
  const handleSaveAndPreview = jest.fn();

  test('should should display form correctly', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <Offerings
            onSubmit={onSubmit}
            user={user}
            validationErrors={validationErrors}
            handleErrorReset={handleErrorReset}
            handleOfferingImageSelection={handleOfferingImageSelection}
            offerings={offerings}
            selectedImage={selectedImage}
            handleSaveAndPreview={handleSaveAndPreview}
          />
        </Router>
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });

  test('should submit the create offering form data', () => {
    const { getByTestId } = render(
      <Provider store={mockStore}>
        <Router>
          <Offerings
            onSubmit={onSubmit}
            user={user}
            validationErrors={validationErrors}
            handleErrorReset={handleErrorReset}
            handleOfferingImageSelection={handleOfferingImageSelection}
            offerings={offerings}
            selectedImage={selectedImage}
            handleSaveAndPreview={handleSaveAndPreview}
          />
        </Router>
      </Provider>
    );

    const saveOfferingBtn = getByTestId('add-offering');
    saveOfferingBtn.onclick = onSubmit;
    fireEvent.click(saveOfferingBtn);

    expect(onSubmit).toHaveBeenCalledTimes(1);
  });
});
