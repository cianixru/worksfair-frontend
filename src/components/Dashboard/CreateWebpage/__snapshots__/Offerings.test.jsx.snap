// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offerings.jsx should should display form correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="margin-bottom-25"
        >
          <ul />
        </div>
        <div
          class="columns box margin-bottom-25"
        >
          <div
            class="column is-one-third"
          >
            <div
              class="image-upload-box"
            >
              <img
                alt=""
                class="offering-image"
                src="dahehjsdsa.ocm"
              />
              <div
                class="change-image-btn"
              >
                <div
                  class="imageUpload"
                  tabindex="0"
                >
                  <span>
                    <i
                      class="fa fa-upload"
                    />
                     Upload Pic
                  </span>
                  <input
                    accept="image/*"
                    autocomplete="off"
                    style="display: none;"
                    tabindex="-1"
                    type="file"
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            class="column"
          >
            <form>
              <div
                class="field"
              >
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="offering-name"
                    name="title"
                    placeholder="Name of item"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="offering-description"
                    name="description"
                    placeholder="Describe this item"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="field"
              >
                <div
                  class="control"
                >
                  <input
                    class="input is-medium"
                    data-testid="offering-price"
                    name="price"
                    placeholder="Price"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="control buttons are-medium"
              >
                <button
                  class="button is-light"
                  data-testid="add-offering"
                  disabled=""
                  type="submit"
                >
                  Add Item
                </button>
                <button
                  class="button"
                  type="button"
                >
                  Reset
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="control"
        >
          <button
            class="button is-medium is-link is-outlined is-rounded"
            data-testid="save-and-preview"
            type="submit"
          >
            Preview Webpage
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="margin-bottom-25"
      >
        <ul />
      </div>
      <div
        class="columns box margin-bottom-25"
      >
        <div
          class="column is-one-third"
        >
          <div
            class="image-upload-box"
          >
            <img
              alt=""
              class="offering-image"
              src="dahehjsdsa.ocm"
            />
            <div
              class="change-image-btn"
            >
              <div
                class="imageUpload"
                tabindex="0"
              >
                <span>
                  <i
                    class="fa fa-upload"
                  />
                   Upload Pic
                </span>
                <input
                  accept="image/*"
                  autocomplete="off"
                  style="display: none;"
                  tabindex="-1"
                  type="file"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="column"
        >
          <form>
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="offering-name"
                  name="title"
                  placeholder="Name of item"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="offering-description"
                  name="description"
                  placeholder="Describe this item"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="control"
              >
                <input
                  class="input is-medium"
                  data-testid="offering-price"
                  name="price"
                  placeholder="Price"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="control buttons are-medium"
            >
              <button
                class="button is-light"
                data-testid="add-offering"
                disabled=""
                type="submit"
              >
                Add Item
              </button>
              <button
                class="button"
                type="button"
              >
                Reset
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="control"
      >
        <button
          class="button is-medium is-link is-outlined is-rounded"
          data-testid="save-and-preview"
          type="submit"
        >
          Preview Webpage
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
